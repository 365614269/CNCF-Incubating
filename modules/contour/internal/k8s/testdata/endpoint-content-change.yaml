apiVersion: v1
kind: Endpoints
metadata:
  annotations:
    endpoints.kubernetes.io/last-change-trigger-time: "2023-02-09T15:06:20Z"
  creationTimestamp: "2023-02-09T14:55:33Z"
  name: echoserver
  namespace: default
  resourceVersion: "85303"
  uid: 6cb4c5f2-60b2-4240-b2a3-91111c7e2527
subsets:
- addresses:
  - ip: 10.244.1.4
    nodeName: contour-worker
    targetRef:
      kind: Pod
      name: echoserver-59db9c5778-cfwhz
      namespace: default
      uid: b0014607-0e3b-461e-8b06-df4407fa7a44
  ports:
  - name: http
    port: 3000
    protocol: TCP
---
apiVersion: v1
kind: Endpoints
metadata:
  annotations:
    endpoints.kubernetes.io/last-change-trigger-time: "2023-02-09T19:56:07Z"
  creationTimestamp: "2023-02-09T14:55:33Z"
  name: echoserver
  namespace: default
  resourceVersion: "112662"
  uid: 6cb4c5f2-60b2-4240-b2a3-91111c7e2527
subsets:
- addresses:
  - ip: 10.244.1.5
    nodeName: contour-worker
    targetRef:
      kind: Pod
      name: echoserver-59db9c5778-5wdhj
      namespace: default
      uid: 84878472-00b6-4ae2-b1fd-5508c900707c
  ports:
  - name: http
    port: 3000
    protocol: TCP
