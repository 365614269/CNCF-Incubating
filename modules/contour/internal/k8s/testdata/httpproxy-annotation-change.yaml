apiVersion: projectcontour.io/v1
kind: HTTPProxy
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"projectcontour.io/v1","kind":"HTTPProxy","metadata":{"annotations":{},"name":"echoserver","namespace":"default"},"spec":{"routes":[{"services":[{"name":"echoserver","port":80}]}],"virtualhost":{"fqdn":"echoserver.127-0-0-101.nip.io"}}}
  creationTimestamp: "2023-02-08T18:32:43Z"
  generation: 1
  name: echoserver
  namespace: default
  resourceVersion: "84327"
  uid: fd31fdfc-bbcd-46c3-af0d-8907da000320
spec:
  routes:
  - services:
    - name: echoserver
      port: 80
  virtualhost:
    fqdn: echoserver.127-0-0-101.nip.io
status:
  conditions:
  - lastTransitionTime: "2023-02-09T14:56:45Z"
    message: Valid HTTPProxy
    observedGeneration: 1
    reason: Valid
    status: "True"
    type: Valid
  currentStatus: valid
  description: Valid HTTPProxy
  loadBalancer: {}
---
apiVersion: projectcontour.io/v1
kind: HTTPProxy
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"projectcontour.io/v1","kind":"HTTPProxy","metadata":{"annotations":{},"name":"echoserver","namespace":"default"},"spec":{"routes":[{"services":[{"name":"echoserver","port":80}]}],"virtualhost":{"fqdn":"echoserver.127-0-0-101.nip.io"}}}
    kubernetes.io/ingress.class: my-ingress
  creationTimestamp: "2023-02-08T18:32:43Z"
  generation: 1
  name: echoserver
  namespace: default
  resourceVersion: "205411"
  uid: fd31fdfc-bbcd-46c3-af0d-8907da000320
spec:
  routes:
  - services:
    - name: echoserver
      port: 80
  virtualhost:
    fqdn: echoserver.127-0-0-101.nip.io
status:
  conditions:
  - lastTransitionTime: "2023-02-09T14:56:45Z"
    message: Valid HTTPProxy
    observedGeneration: 1
    reason: Valid
    status: "True"
    type: Valid
  currentStatus: valid
  description: Valid HTTPProxy
  loadBalancer: {}
