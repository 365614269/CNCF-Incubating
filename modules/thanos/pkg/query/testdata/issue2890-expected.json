{
  "status": "success",
  "data": {
    "resultType": "matrix",
    "result": [
      {
        "metric": {
          "__name__": "cluster_version",
          "_id": "014d6c9f-9c0f-40e2-8ab9-73ae46dbc9b1",
          "replica": "b"
        },
        "values": [
          [
            1598471700,
            "1596610196"
          ],
          [
            1598471703,
            "1596610196"
          ],
          [
            1598471706,
            "1596610196"
          ],
          [
            1598471709,
            "1596610196"
          ],
          [
            1598471712,
            "1596610196"
          ],
          [
            1598471715,
            "1596610196"
          ],
          [
            1598471718,
            "1596610196"
          ],
          [
            1598471721,
            "1596610196"
          ],
          [
            1598471724,
            "1596610196"
          ],
          [
            1598471727,
            "1596610196"
          ],
          [
            1598471730,
            "1596610196"
          ],
          [
            1598471733,
            "1596610196"
          ],
          [
            1598471736,
            "1596610196"
          ],
          [
            1598471739,
            "1596610196"
          ],
          [
            1598471742,
            "1596610196"
          ],
          [
            1598471745,
            "1596610196"
          ],
          [
            1598471748,
            "1596610196"
          ],
          [
            1598471751,
            "1596610196"
          ],
          [
            1598471754,
            "1596610196"
          ],
          [
            1598471757,
            "1596610196"
          ],
          [
            1598471760,
            "1596610196"
          ],
          [
            1598471763,
            "1596610196"
          ],
          [
            1598471766,
            "1596610196"
          ],
          [
            1598471769,
            "1596610196"
          ],
          [
            1598471772,
            "1596610196"
          ],
          [
            1598471775,
            "1596610196"
          ],
          [
            1598471778,
            "1596610196"
          ],
          [
            1598471781,
            "1596610196"
          ],
          [
            1598471784,
            "1596610196"
          ],
          [
            1598471787,
            "1596610196"
          ],
          [
            1598471790,
            "1596610196"
          ],
          [
            1598471793,
            "1596610196"
          ],
          [
            1598471796,
            "1596610196"
          ],
          [
            1598471799,
            "1596610196"
          ],
          [
            1598471802,
            "1596610196"
          ],
          [
            1598471805,
            "1596610196"
          ],
          [
            1598471808,
            "1596610196"
          ],
          [
            1598471811,
            "1596610196"
          ],
          [
            1598471814,
            "1596610196"
          ],
          [
            1598471817,
            "1596610196"
          ],
          [
            1598471820,
            "1596610196"
          ],
          [
            1598471823,
            "1596610196"
          ],
          [
            1598471826,
            "1596610196"
          ],
          [
            1598471829,
            "1596610196"
          ],
          [
            1598471832,
            "1596610196"
          ],
          [
            1598471835,
            "1596610196"
          ],
          [
            1598471838,
            "1596610196"
          ],
          [
            1598471841,
            "1596610196"
          ],
          [
            1598471844,
            "1596610196"
          ],
          [
            1598471847,
            "1596610196"
          ],
          [
            1598471850,
            "1596610196"
          ],
          [
            1598471853,
            "1596610196"
          ],
          [
            1598471856,
            "1596610196"
          ],
          [
            1598471859,
            "1596610196"
          ],
          [
            1598471862,
            "1596610196"
          ],
          [
            1598471865,
            "1596610196"
          ],
          [
            1598471868,
            "1596610196"
          ],
          [
            1598471871,
            "1596610196"
          ],
          [
            1598471874,
            "1596610196"
          ],
          [
            1598471877,
            "1596610196"
          ],
          [
            1598471880,
            "1596610196"
          ],
          [
            1598471883,
            "1596610196"
          ],
          [
            1598471886,
            "1596610196"
          ],
          [
            1598471889,
            "1596610196"
          ],
          [
            1598471892,
            "1596610196"
          ],
          [
            1598471895,
            "1596610196"
          ],
          [
            1598471898,
            "1596610196"
          ],
          [
            1598471901,
            "1596610196"
          ],
          [
            1598471904,
            "1596610196"
          ],
          [
            1598471907,
            "1596610196"
          ],
          [
            1598471910,
            "1596610196"
          ],
          [
            1598471913,
            "1596610196"
          ],
          [
            1598471916,
            "1596610196"
          ],
          [
            1598471919,
            "1596610196"
          ],
          [
            1598471922,
            "1596610196"
          ],
          [
            1598471925,
            "1596610196"
          ],
          [
            1598471928,
            "1596610196"
          ],
          [
            1598471931,
            "1596610196"
          ],
          [
            1598471934,
            "1596610196"
          ],
          [
            1598471937,
            "1596610196"
          ],
          [
            1598471940,
            "1596610196"
          ],
          [
            1598471943,
            "1596610196"
          ],
          [
            1598471946,
            "1596610196"
          ],
          [
            1598471949,
            "1596610196"
          ],
          [
            1598471952,
            "1596610196"
          ],
          [
            1598471955,
            "1596610196"
          ],
          [
            1598471958,
            "1596610196"
          ],
          [
            1598471961,
            "1596610196"
          ],
          [
            1598471964,
            "1596610196"
          ],
          [
            1598471967,
            "1596610196"
          ],
          [
            1598471970,
            "1596610196"
          ],
          [
            1598471973,
            "1596610196"
          ],
          [
            1598471976,
            "1596610196"
          ],
          [
            1598471979,
            "1596610196"
          ],
          [
            1598471982,
            "1596610196"
          ],
          [
            1598471985,
            "1596610196"
          ],
          [
            1598471988,
            "1596610196"
          ],
          [
            1598471991,
            "1596610196"
          ],
          [
            1598471994,
            "1596610196"
          ],
          [
            1598471997,
            "1596610196"
          ],
          [
            1598472000,
            "1596610196"
          ],
          [
            1598472003,
            "1596610196"
          ],
          [
            1598472006,
            "1596610196"
          ],
          [
            1598472009,
            "1596610196"
          ],
          [
            1598472012,
            "1596610196"
          ],
          [
            1598472015,
            "1596610196"
          ],
          [
            1598472018,
            "1596610196"
          ],
          [
            1598472021,
            "1596610196"
          ],
          [
            1598472024,
            "1596610196"
          ],
          [
            1598472027,
            "1596610196"
          ],
          [
            1598472030,
            "1596610196"
          ],
          [
            1598472033,
            "1596610196"
          ],
          [
            1598472036,
            "1596610196"
          ],
          [
            1598472039,
            "1596610196"
          ],
          [
            1598472042,
            "1596610196"
          ],
          [
            1598472045,
            "1596610196"
          ],
          [
            1598472048,
            "1596610196"
          ],
          [
            1598472051,
            "1596610196"
          ],
          [
            1598472054,
            "1596610196"
          ],
          [
            1598472057,
            "1596610196"
          ],
          [
            1598472060,
            "1596610196"
          ],
          [
            1598472063,
            "1596610196"
          ],
          [
            1598472066,
            "1596610196"
          ],
          [
            1598472069,
            "1596610196"
          ],
          [
            1598472072,
            "1596610196"
          ],
          [
            1598472075,
            "1596610196"
          ],
          [
            1598472078,
            "1596610196"
          ],
          [
            1598472081,
            "1596610196"
          ],
          [
            1598472084,
            "1596610196"
          ],
          [
            1598472087,
            "1596610196"
          ],
          [
            1598472090,
            "1596610196"
          ],
          [
            1598472093,
            "1596610196"
          ],
          [
            1598472096,
            "1596610196"
          ],
          [
            1598472099,
            "1596610196"
          ],
          [
            1598472102,
            "1596610196"
          ],
          [
            1598472105,
            "1596610196"
          ],
          [
            1598472108,
            "1596610196"
          ],
          [
            1598472111,
            "1596610196"
          ],
          [
            1598472114,
            "1596610196"
          ],
          [
            1598472117,
            "1596610196"
          ],
          [
            1598472120,
            "1596610196"
          ],
          [
            1598472123,
            "1596610196"
          ],
          [
            1598472126,
            "1596610196"
          ],
          [
            1598472129,
            "1596610196"
          ],
          [
            1598472132,
            "1596610196"
          ],
          [
            1598472135,
            "1596610196"
          ],
          [
            1598472138,
            "1596610196"
          ],
          [
            1598472141,
            "1596610196"
          ],
          [
            1598472144,
            "1596610196"
          ],
          [
            1598472147,
            "1596610196"
          ],
          [
            1598472150,
            "1596610196"
          ],
          [
            1598472153,
            "1596610196"
          ],
          [
            1598472156,
            "1596610196"
          ],
          [
            1598472159,
            "1596610196"
          ],
          [
            1598472162,
            "1596610196"
          ],
          [
            1598472165,
            "1596610196"
          ],
          [
            1598472168,
            "1596610196"
          ],
          [
            1598472171,
            "1596610196"
          ],
          [
            1598472174,
            "1596610196"
          ],
          [
            1598472177,
            "1596610196"
          ],
          [
            1598472180,
            "1596610196"
          ],
          [
            1598472183,
            "1596610196"
          ],
          [
            1598472186,
            "1596610196"
          ],
          [
            1598472189,
            "1596610196"
          ],
          [
            1598472192,
            "1596610196"
          ],
          [
            1598472195,
            "1596610196"
          ],
          [
            1598472198,
            "1596610196"
          ],
          [
            1598472201,
            "1596610196"
          ],
          [
            1598472204,
            "1596610196"
          ],
          [
            1598472207,
            "1596610196"
          ],
          [
            1598472210,
            "1596610196"
          ],
          [
            1598472213,
            "1596610196"
          ],
          [
            1598472216,
            "1596610196"
          ],
          [
            1598472219,
            "1596610196"
          ],
          [
            1598472222,
            "1596610196"
          ],
          [
            1598472225,
            "1596610196"
          ],
          [
            1598472228,
            "1596610196"
          ],
          [
            1598472231,
            "1596610196"
          ],
          [
            1598472234,
            "1596610196"
          ],
          [
            1598472237,
            "1596610196"
          ],
          [
            1598472240,
            "1596610196"
          ],
          [
            1598472243,
            "1596610196"
          ],
          [
            1598472246,
            "1596610196"
          ],
          [
            1598472249,
            "1596610196"
          ],
          [
            1598472252,
            "1596610196"
          ],
          [
            1598472255,
            "1596610196"
          ],
          [
            1598472258,
            "1596610196"
          ],
          [
            1598472261,
            "1596610196"
          ],
          [
            1598472264,
            "1596610196"
          ],
          [
            1598472267,
            "1596610196"
          ],
          [
            1598472270,
            "1596610196"
          ],
          [
            1598472273,
            "1596610196"
          ],
          [
            1598472276,
            "1596610196"
          ],
          [
            1598472279,
            "1596610196"
          ],
          [
            1598472282,
            "1596610196"
          ],
          [
            1598472285,
            "1596610196"
          ],
          [
            1598472288,
            "1596610196"
          ],
          [
            1598472291,
            "1596610196"
          ],
          [
            1598472294,
            "1596610196"
          ],
          [
            1598472297,
            "1596610196"
          ],
          [
            1598472300,
            "1596610196"
          ],
          [
            1598472303,
            "1596610196"
          ],
          [
            1598472306,
            "1596610196"
          ],
          [
            1598472309,
            "1596610196"
          ],
          [
            1598472312,
            "1596610196"
          ],
          [
            1598472315,
            "1596610196"
          ],
          [
            1598472318,
            "1596610196"
          ],
          [
            1598472321,
            "1596610196"
          ],
          [
            1598472324,
            "1596610196"
          ],
          [
            1598472327,
            "1596610196"
          ],
          [
            1598472330,
            "1596610196"
          ],
          [
            1598472333,
            "1596610196"
          ],
          [
            1598472336,
            "1596610196"
          ],
          [
            1598472339,
            "1596610196"
          ],
          [
            1598472342,
            "1596610196"
          ],
          [
            1598472345,
            "1596610196"
          ],
          [
            1598472348,
            "1596610196"
          ],
          [
            1598472351,
            "1596610196"
          ],
          [
            1598472354,
            "1596610196"
          ],
          [
            1598472357,
            "1596610196"
          ],
          [
            1598472360,
            "1596610196"
          ],
          [
            1598472363,
            "1596610196"
          ],
          [
            1598472366,
            "1596610196"
          ],
          [
            1598472369,
            "1596610196"
          ],
          [
            1598472372,
            "1596610196"
          ],
          [
            1598472375,
            "1596610196"
          ],
          [
            1598472378,
            "1596610196"
          ],
          [
            1598472381,
            "1596610196"
          ],
          [
            1598472384,
            "1596610196"
          ],
          [
            1598472387,
            "1596610196"
          ],
          [
            1598472390,
            "1596610196"
          ],
          [
            1598472393,
            "1596610196"
          ],
          [
            1598472396,
            "1596610196"
          ],
          [
            1598472399,
            "1596610196"
          ],
          [
            1598472402,
            "1596610196"
          ],
          [
            1598472405,
            "1596610196"
          ],
          [
            1598472408,
            "1596610196"
          ],
          [
            1598472411,
            "1596610196"
          ],
          [
            1598472414,
            "1596610196"
          ],
          [
            1598472417,
            "1596610196"
          ],
          [
            1598472420,
            "1596610196"
          ],
          [
            1598472423,
            "1596610196"
          ],
          [
            1598472426,
            "1596610196"
          ],
          [
            1598472429,
            "1596610196"
          ],
          [
            1598472432,
            "1596610196"
          ],
          [
            1598472435,
            "1596610196"
          ],
          [
            1598472438,
            "1596610196"
          ],
          [
            1598472441,
            "1596610196"
          ],
          [
            1598472444,
            "1596610196"
          ],
          [
            1598472447,
            "1596610196"
          ],
          [
            1598472450,
            "1596610196"
          ],
          [
            1598472453,
            "1596610196"
          ],
          [
            1598472456,
            "1596610196"
          ],
          [
            1598472459,
            "1596610196"
          ],
          [
            1598472462,
            "1596610196"
          ],
          [
            1598472465,
            "1596610196"
          ],
          [
            1598472468,
            "1596610196"
          ],
          [
            1598472471,
            "1596610196"
          ],
          [
            1598472474,
            "1596610196"
          ],
          [
            1598472477,
            "1596610196"
          ],
          [
            1598472480,
            "1596610196"
          ],
          [
            1598472483,
            "1596610196"
          ],
          [
            1598472486,
            "1596610196"
          ],
          [
            1598472489,
            "1596610196"
          ],
          [
            1598472492,
            "1596610196"
          ],
          [
            1598472495,
            "1596610196"
          ],
          [
            1598472498,
            "1596610196"
          ],
          [
            1598472501,
            "1596610196"
          ],
          [
            1598472504,
            "1596610196"
          ],
          [
            1598472507,
            "1596610196"
          ],
          [
            1598472510,
            "1596610196"
          ],
          [
            1598472513,
            "1596610196"
          ],
          [
            1598472516,
            "1596610196"
          ],
          [
            1598472519,
            "1596610196"
          ],
          [
            1598472522,
            "1596610196"
          ],
          [
            1598472525,
            "1596610196"
          ],
          [
            1598472528,
            "1596610196"
          ],
          [
            1598472531,
            "1596610196"
          ],
          [
            1598472534,
            "1596610196"
          ],
          [
            1598472537,
            "1596610196"
          ],
          [
            1598472540,
            "1596610196"
          ],
          [
            1598472543,
            "1596610196"
          ],
          [
            1598472546,
            "1596610196"
          ],
          [
            1598472549,
            "1596610196"
          ],
          [
            1598472552,
            "1596610196"
          ],
          [
            1598472555,
            "1596610196"
          ],
          [
            1598472558,
            "1596610196"
          ],
          [
            1598472561,
            "1596610196"
          ],
          [
            1598472564,
            "1596610196"
          ],
          [
            1598472567,
            "1596610196"
          ],
          [
            1598472570,
            "1596610196"
          ],
          [
            1598472573,
            "1596610196"
          ],
          [
            1598472576,
            "1596610196"
          ],
          [
            1598472579,
            "1596610196"
          ],
          [
            1598472582,
            "1596610196"
          ],
          [
            1598472585,
            "1596610196"
          ],
          [
            1598472588,
            "1596610196"
          ],
          [
            1598472591,
            "1596610196"
          ],
          [
            1598472594,
            "1596610196"
          ],
          [
            1598472597,
            "1596610196"
          ],
          [
            1598472600,
            "1596610196"
          ]
        ]
      },
      {
        "metric": {
          "__name__": "cluster_version",
          "_id": "014d6c9f-9c0f-40e2-8ab9-73ae46dbc9b1",
          "replica": "a"
        },
        "values": [
          [
            1598472162,
            "1596610196"
          ],
          [
            1598472165,
            "1596610196"
          ],
          [
            1598472168,
            "1596610196"
          ],
          [
            1598472171,
            "1596610196"
          ],
          [
            1598472174,
            "1596610196"
          ],
          [
            1598472177,
            "1596610196"
          ],
          [
            1598472180,
            "1596610196"
          ],
          [
            1598472183,
            "1596610196"
          ],
          [
            1598472186,
            "1596610196"
          ],
          [
            1598472189,
            "1596610196"
          ],
          [
            1598472192,
            "1596610196"
          ],
          [
            1598472195,
            "1596610196"
          ],
          [
            1598472198,
            "1596610196"
          ],
          [
            1598472201,
            "1596610196"
          ],
          [
            1598472204,
            "1596610196"
          ],
          [
            1598472207,
            "1596610196"
          ],
          [
            1598472210,
            "1596610196"
          ],
          [
            1598472213,
            "1596610196"
          ],
          [
            1598472216,
            "1596610196"
          ],
          [
            1598472219,
            "1596610196"
          ],
          [
            1598472222,
            "1596610196"
          ],
          [
            1598472225,
            "1596610196"
          ],
          [
            1598472228,
            "1596610196"
          ],
          [
            1598472231,
            "1596610196"
          ],
          [
            1598472234,
            "1596610196"
          ],
          [
            1598472237,
            "1596610196"
          ],
          [
            1598472240,
            "1596610196"
          ],
          [
            1598472243,
            "1596610196"
          ],
          [
            1598472246,
            "1596610196"
          ],
          [
            1598472249,
            "1596610196"
          ],
          [
            1598472252,
            "1596610196"
          ],
          [
            1598472255,
            "1596610196"
          ],
          [
            1598472258,
            "1596610196"
          ],
          [
            1598472261,
            "1596610196"
          ],
          [
            1598472264,
            "1596610196"
          ],
          [
            1598472267,
            "1596610196"
          ],
          [
            1598472270,
            "1596610196"
          ],
          [
            1598472273,
            "1596610196"
          ],
          [
            1598472276,
            "1596610196"
          ],
          [
            1598472279,
            "1596610196"
          ],
          [
            1598472282,
            "1596610196"
          ],
          [
            1598472285,
            "1596610196"
          ],
          [
            1598472288,
            "1596610196"
          ],
          [
            1598472291,
            "1596610196"
          ],
          [
            1598472294,
            "1596610196"
          ],
          [
            1598472297,
            "1596610196"
          ],
          [
            1598472300,
            "1596610196"
          ],
          [
            1598472303,
            "1596610196"
          ],
          [
            1598472306,
            "1596610196"
          ],
          [
            1598472309,
            "1596610196"
          ],
          [
            1598472312,
            "1596610196"
          ],
          [
            1598472315,
            "1596610196"
          ],
          [
            1598472318,
            "1596610196"
          ],
          [
            1598472321,
            "1596610196"
          ],
          [
            1598472324,
            "1596610196"
          ],
          [
            1598472327,
            "1596610196"
          ],
          [
            1598472330,
            "1596610196"
          ],
          [
            1598472333,
            "1596610196"
          ],
          [
            1598472336,
            "1596610196"
          ],
          [
            1598472339,
            "1596610196"
          ],
          [
            1598472342,
            "1596610196"
          ],
          [
            1598472345,
            "1596610196"
          ],
          [
            1598472348,
            "1596610196"
          ],
          [
            1598472351,
            "1596610196"
          ],
          [
            1598472354,
            "1596610196"
          ],
          [
            1598472357,
            "1596610196"
          ],
          [
            1598472360,
            "1596610196"
          ],
          [
            1598472363,
            "1596610196"
          ],
          [
            1598472366,
            "1596610196"
          ],
          [
            1598472369,
            "1596610196"
          ],
          [
            1598472372,
            "1596610196"
          ],
          [
            1598472375,
            "1596610196"
          ],
          [
            1598472378,
            "1596610196"
          ],
          [
            1598472381,
            "1596610196"
          ],
          [
            1598472384,
            "1596610196"
          ],
          [
            1598472387,
            "1596610196"
          ],
          [
            1598472390,
            "1596610196"
          ],
          [
            1598472393,
            "1596610196"
          ],
          [
            1598472396,
            "1596610196"
          ],
          [
            1598472399,
            "1596610196"
          ],
          [
            1598472402,
            "1596610196"
          ],
          [
            1598472405,
            "1596610196"
          ],
          [
            1598472408,
            "1596610196"
          ],
          [
            1598472411,
            "1596610196"
          ],
          [
            1598472414,
            "1596610196"
          ],
          [
            1598472417,
            "1596610196"
          ],
          [
            1598472420,
            "1596610196"
          ],
          [
            1598472423,
            "1596610196"
          ],
          [
            1598472426,
            "1596610196"
          ],
          [
            1598472429,
            "1596610196"
          ],
          [
            1598472432,
            "1596610196"
          ],
          [
            1598472435,
            "1596610196"
          ],
          [
            1598472438,
            "1596610196"
          ],
          [
            1598472441,
            "1596610196"
          ],
          [
            1598472444,
            "1596610196"
          ],
          [
            1598472447,
            "1596610196"
          ],
          [
            1598472450,
            "1596610196"
          ],
          [
            1598472453,
            "1596610196"
          ],
          [
            1598472456,
            "1596610196"
          ],
          [
            1598472459,
            "1596610196"
          ],
          [
            1598472462,
            "1596610196"
          ],
          [
            1598472465,
            "1596610196"
          ],
          [
            1598472468,
            "1596610196"
          ],
          [
            1598472471,
            "1596610196"
          ],
          [
            1598472474,
            "1596610196"
          ],
          [
            1598472477,
            "1596610196"
          ],
          [
            1598472480,
            "1596610196"
          ],
          [
            1598472483,
            "1596610196"
          ],
          [
            1598472486,
            "1596610196"
          ],
          [
            1598472489,
            "1596610196"
          ],
          [
            1598472492,
            "1596610196"
          ],
          [
            1598472495,
            "1596610196"
          ],
          [
            1598472498,
            "1596610196"
          ],
          [
            1598472501,
            "1596610196"
          ],
          [
            1598472504,
            "1596610196"
          ],
          [
            1598472507,
            "1596610196"
          ],
          [
            1598472510,
            "1596610196"
          ],
          [
            1598472513,
            "1596610196"
          ],
          [
            1598472516,
            "1596610196"
          ],
          [
            1598472519,
            "1596610196"
          ],
          [
            1598472522,
            "1596610196"
          ],
          [
            1598472525,
            "1596610196"
          ],
          [
            1598472528,
            "1596610196"
          ],
          [
            1598472531,
            "1596610196"
          ],
          [
            1598472534,
            "1596610196"
          ],
          [
            1598472537,
            "1596610196"
          ],
          [
            1598472540,
            "1596610196"
          ],
          [
            1598472543,
            "1596610196"
          ],
          [
            1598472546,
            "1596610196"
          ],
          [
            1598472549,
            "1596610196"
          ],
          [
            1598472552,
            "1596610196"
          ],
          [
            1598472555,
            "1596610196"
          ],
          [
            1598472558,
            "1596610196"
          ],
          [
            1598472561,
            "1596610196"
          ],
          [
            1598472564,
            "1596610196"
          ],
          [
            1598472567,
            "1596610196"
          ],
          [
            1598472570,
            "1596610196"
          ],
          [
            1598472573,
            "1596610196"
          ],
          [
            1598472576,
            "1596610196"
          ],
          [
            1598472579,
            "1596610196"
          ],
          [
            1598472582,
            "1596610196"
          ],
          [
            1598472585,
            "1596610196"
          ],
          [
            1598472588,
            "1596610196"
          ],
          [
            1598472591,
            "1596610196"
          ],
          [
            1598472594,
            "1596610196"
          ],
          [
            1598472597,
            "1596610196"
          ],
          [
            1598472600,
            "1596610196"
          ]
        ]
      }
    ]
  }
}