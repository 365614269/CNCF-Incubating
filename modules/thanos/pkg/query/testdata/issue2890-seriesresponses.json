{"series":{"labels":[{"name":"__name__","value":"cluster_version"},{"name":"_id","value":"014d6c9f-9c0f-40e2-8ab9-73ae46dbc9b1"},{"name":"replica","value":"b"}],"chunks":[{"minTime":"1598464999000","maxTime":"1598471891000","raw":{"data":"ABqw+YG/hV1B18qVpQAAAJjFEAF4MIPoAGdTA0VoAZzPDRmIAF4MIPoQfQvBheDA"}}]}}
{"series":{"labels":[{"name":"__name__","value":"cluster_version"},{"name":"_id","value":"014d6c9f-9c0f-40e2-8ab9-73ae46dbc9b1"},{"name":"replica","value":"a"}],"chunks":[{"minTime":"1598472162000","maxTime":"1598478935000","raw":{"data":"ABqgq+zFhV1B18qVpQAAAJjFEABeDCD6AABeDCD6ABeDCD6AQfQA"}}]}}
