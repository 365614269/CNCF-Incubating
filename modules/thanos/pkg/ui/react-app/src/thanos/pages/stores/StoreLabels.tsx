import React, { FC } from 'react';
import { Badge, ListGroup, ListGroupItem } from 'reactstrap';
import { Labels } from './store';

export type StoreLabelsProps = { labelSets: Labels[] };

export const StoreLabels: FC<StoreLabelsProps> = ({ labelSets }) => {
  return (
    <ListGroup>
      {labelSets.map((labels, idx) => (
        <ListGroupItem key={idx}>
          {Object.entries(labels).map(([name, value]) => (
            <Badge key={name} color="primary" style={{ margin: '0px 5px' }}>{`${name}="${value}"`}</Badge>
          ))}
        </ListGroupItem>
      ))}
    </ListGroup>
  );
};

export default StoreLabels;
