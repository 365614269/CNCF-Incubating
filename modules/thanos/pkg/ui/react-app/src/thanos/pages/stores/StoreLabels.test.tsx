import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { ListGroup, ListGroupItem } from 'reactstrap';
import StoreLabels from './StoreLabels';
import { sampleAPIResponse } from './__testdata__/testdata';

describe('storeLabels', () => {
  const { labelSets } = sampleAPIResponse.data.store[0];
  const storeLabels = shallow(<StoreLabels labelSets={labelSets} />);

  it('renders a listGroup', () => {
    const listGroup = storeLabels.find(ListGroup);
    expect(listGroup).toHaveLength(1);
  });

  it('renders a ListGroupItem for each labelSet', () => {
    const listGroupItems = storeLabels.find(ListGroupItem);
    expect(listGroupItems).toHaveLength(labelSets.length);
  });

  it('renders discovered labels', () => {
    expect(toJson(storeLabels)).toMatchSnapshot();
  });
});
