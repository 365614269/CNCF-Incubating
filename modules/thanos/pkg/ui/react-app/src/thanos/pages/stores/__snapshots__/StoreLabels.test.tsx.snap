// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`storeLabels renders discovered labels 1`] = `
<ListGroup
  horizontal={false}
  tag="ul"
>
  <ListGroupItem
    key="0"
    tag="li"
  >
    <Badge
      color="primary"
      key="monitor"
      pill={false}
      style={
        Object {
          "margin": "0px 5px",
        }
      }
      tag="span"
    >
      monitor="prometheus_one"
    </Badge>
  </ListGroupItem>
  <ListGroupItem
    key="1"
    tag="li"
  >
    <Badge
      color="primary"
      key="monitor"
      pill={false}
      style={
        Object {
          "margin": "0px 5px",
        }
      }
      tag="span"
    >
      monitor="prometheus_one"
    </Badge>
    <Badge
      color="primary"
      key="source"
      pill={false}
      style={
        Object {
          "margin": "0px 5px",
        }
      }
      tag="span"
    >
      source="Thanos"
    </Badge>
  </ListGroupItem>
  <ListGroupItem
    key="2"
    tag="li"
  >
    <Badge
      color="primary"
      key="monitor"
      pill={false}
      style={
        Object {
          "margin": "0px 5px",
        }
      }
      tag="span"
    >
      monitor="prometheus_one"
    </Badge>
    <Badge
      color="primary"
      key="source"
      pill={false}
      style={
        Object {
          "margin": "0px 5px",
        }
      }
      tag="span"
    >
      source="Thanos1"
    </Badge>
  </ListGroupItem>
  <ListGroupItem
    key="3"
    tag="li"
  >
    <Badge
      color="primary"
      key="monitor"
      pill={false}
      style={
        Object {
          "margin": "0px 5px",
        }
      }
      tag="span"
    >
      monitor="prometheus_two"
    </Badge>
  </ListGroupItem>
  <ListGroupItem
    key="4"
    tag="li"
  >
    <Badge
      color="primary"
      key="monitor"
      pill={false}
      style={
        Object {
          "margin": "0px 5px",
        }
      }
      tag="span"
    >
      monitor="prometheus_two"
    </Badge>
    <Badge
      color="primary"
      key="source"
      pill={false}
      style={
        Object {
          "margin": "0px 5px",
        }
      }
      tag="span"
    >
      source="Thanos"
    </Badge>
  </ListGroupItem>
  <ListGroupItem
    key="5"
    tag="li"
  >
    <Badge
      color="primary"
      key="monitor"
      pill={false}
      style={
        Object {
          "margin": "0px 5px",
        }
      }
      tag="span"
    >
      monitor="prometheus_two"
    </Badge>
    <Badge
      color="primary"
      key="source"
      pill={false}
      style={
        Object {
          "margin": "0px 5px",
        }
      }
      tag="span"
    >
      source="Thanos2"
    </Badge>
  </ListGroupItem>
</ListGroup>
`;
